BoxContentSDK/BoxContentSDK/Operations/BOXAPIOperation.m

```diff
@@ -205,6 +205,23 @@ - (void)prepareAPIRequest
     BOXAbstract();
 }
 
+- (void)prepareOperation
+{
+    if (![self isCancelled]) {
+        if (self.sessionTask == nil) {
+            @synchronized(self.session)
+            {
+                //Note: if sessionTask exists, we cannot change its API request
+                //make sure you recreate sessionTask with the new API request if needed
+                [self prepareAPIRequest];
+                self.accessToken = self.session.accessToken;
+            }
+            NSError *error = nil;
+            self.sessionTask = [self createSessionTaskWithError:&error];
```

Comment:
if there's an error creating session task, we should execute logic of line 374 to 377 to set self.error accordingly. We potentially have a valid self.sessionTask but there's other error setting up session task (in background tasks, we need to set up its cache before the task starts)